File: widgetssdk/src/main/java/com/glia/widgets/fileupload/domain/RemoveFileAttachmentUseCase.java
package com.glia.widgets.fileupload.domain;

import com.glia.widgets.fileupload.FileAttachmentRepository;
import com.glia.widgets.fileupload.model.FileAttachment;

public class RemoveFileAttachmentUseCase {
    private final FileAttachmentRepository repository;

    public RemoveFileAttachmentUseCase(FileAttachmentRepository repository) {
        this.repository = repository;
    }

    public void execute(FileAttachment attachment) {
        repository.detachFile(attachment);
    }
}

File: widgetssdk/src/main/java/com/glia/widgets/internal/fileupload/domain/RemoveFileAttachmentUseCase.kt
package com.glia.widgets.internal.fileupload.domain

import com.glia.widgets.internal.fileupload.FileAttachmentRepository
import com.glia.widgets.internal.fileupload.model.LocalAttachment

internal class RemoveFileAttachmentUseCase(private val repository: FileAttachmentRepository) {
    operator fun invoke(attachment: LocalAttachment) {
        repository.detachFile(attachment)
    }
}
